### PR TITLE
Hosts in a delegation get zone = None, not the parent zone.

### LINKED ISSUE #362: Hosts in a delegation shall get zone == None, not the parentzone
E.g with zone `example.org` and delegation `sub.example.org` when host `hostname.sub.example.org` is created the `zone` attribute must be `None` and not point to  `example.org`.

Make sure that glue records for delegations still are included in the parentzone and are now at the top, where they should be.

### PR DESCRIPTION
**Pull request by oyvindhagberg**
Fixes #362.

### CODE REVIEW

File: mreg/signals.py [hunk 1]
```diff
@@ -298,3 +299,13 @@ def update_hosts_when_zone_is_added(sender, instance, created, **kwargs):
                 continue
             h.zone = instance
             h.save()
+
+@receiver(pre_save, sender=Host)
+def hosts_in_a_delegation_shall_get_zone_None(sender, instance, raw, using, update_fields, **kwargs):
+    """ Hosts in a delegation shall get zone == None, not the parentzone. #362 """
+    if instance.zone != None:
+        for d in ForwardZoneDelegation.objects.filter(zone = instance.zone):
+            if instance.name.endswith('.'+d.name):
```
Comment (oyvindkolbu) [inline mreg/signals.py:308]: Does not handle an instance with the same name as the delegation.

File: mreg/signals.py [hunk 2]
```diff
@@ -298,3 +299,13 @@ def update_hosts_when_zone_is_added(sender, instance, created, **kwargs):
                 continue
             h.zone = instance
             h.save()
+
+@receiver(pre_save, sender=Host)
+def hosts_in_a_delegation_shall_get_zone_None(sender, instance, raw, using, update_fields, **kwargs):
```
Comment (oyvindkolbu) [inline mreg/signals.py:304]: Will slow down things massively. Only needed for new instances or updated name, though not sure if it should be done as a dedicated signal at all.

**Review by mbakke — APPROVED** [review]
:shipit:

**Review by oyvindkolbu — APPROVED** [review]
:shipit:

File: mreg/api/v1/tests/tests_zones.py [hunk 3]
```diff
@@ -366,6 +366,26 @@ def test_delegation_patch_modifies_zone_updated_at(self):
...
+                "ipaddress": "10.10.0.1", "contact": "mail@delegated.example.org"})
+        # load the host object and verify that its zone is None
+        host = Host.objects.get(name="foo.delegated.example.org")
+        self.assertTrue(host.zone is None)
+        # add a host with the same name as the delegation
+        self.assert_post_and_201('/hosts/', {"name": "delegated.example.org",
+                "ipaddress": "10.10.0.2", "contact": "mail@delegated.example.org"})
+        # load the host object and verify that its zone is None
+        host = Host.objects.get(name="delegated.example.org")
+        self.assertTrue(host.zone is None)
```
Comment (oyvindkolbu) [inline mreg/api/v1/tests/tests_zones.py:387]: This is prettier handled with a `self.assertIsNone(host.zone)`.

Also multiple variantions of a test within a test function should use subtesting. See
https://docs.python.org/3/library/unittest.html#distinguishing-test-iterations-using-subtests

Comment (oyvindhagberg) [reply mreg/api/v1/tests/tests_zones.py]: Just added a commit (bb326f97c92ce3ed8ca1c09c8d5102606a551ab7) with these modifications.

**Review by mbakke — APPROVED** [review]

**Pull request merged**